### PR TITLE
Fix plotting VOIs

### LINKED ISSUE #433: Displaying VOI contours causes a crash.
image

Tested in 3.7 and 3.8 versions of Python.

### LINKED ISSUE #435: Implement VOIs plotting in new plotter
Insert VOIs plotting responsibility into plotter module, instead of plotting it somewhere else.

### LINKED ISSUE #461: Bug: Plotting VOIs in sagittal and coronal
Plotting contours in sagittal and coronal resizes image on which those are plotted, see video below:

https://user-images.githubusercontent.com/44035398/131037497-a54ca294-1d39-4171-bfd5-8badcb50bdf5.mp4

see also:
https://github.com/pytrip/pytripgui/pull/437#issuecomment-904613966

### PR DESCRIPTION
**Pull request by rozchlastywacz**
I reworked code that was used to plot VOIs - now it is under plotter scope of responsibilities

Test it, try to find as many bugs you can!

Note: works ONLY on TRANSVERSAL planes

OUTDATED Edit: work in all planes, but plotting in planes other than transversal causes image resizing, for more information see: 
- #461

Edit2: reworking code that plotted POIs now is a separate issue - see: 
- #462

Edit3: some other issues (needed more or less to fix VOI plotting) to be solved: 
- pytrip/pytrip#590 
- pytrip/pytrip#592 
- pytrip/pytrip#593
- #458 
- #448
- #453

Edit4: @michal367 fixed 
- #461, so now it wroks in every plane

closes #461

closes #435
closes #433

### CODE REVIEW

File: pytripgui/canvas_vc/plotter/managers/voi_manager.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    def _get_current_slice(self, vdx, voi):
+        _slice = None
+        # TODO create enum class that holds all plane strings
+        if vdx.projection_selector.plane == "Transversal":
+            _slice = voi.get_slice_at_pos(vdx.ctx.slice_to_z(vdx.projection_selector.current_slice_no + 1))
+        elif vdx.projection_selector.plane == "Sagittal":
+            _slice = voi.get_2d_slice(voi.sagittal, vdx.projection_selector.current_slice_no)
+        elif vdx.projection_selector.plane == "Coronal":
+            _slice = voi.get_2d_slice(voi.coronal, vdx.projection_selector.current_slice_no)
+        return _slice
```
Comment (grzanka) [inline pytripgui/canvas_vc/plotter/managers/voi_manager.py:82]: Add annotation mentioning type of `_slice`

Comment (rozchlastywacz) [reply pytripgui/canvas_vc/plotter/managers/voi_manager.py:96]: done

File: pytripgui/canvas_vc/plotter/managers/voi_manager.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+logger = logging.getLogger(__name__)
+
+
+class VoiManager:
+    def __init__(self, axes: Axes, blit_manager: BlitManager):
+        self._axes: Axes = axes
+        self._blit_manager: BlitManager = blit_manager
+        self._plotted_voi: dict = {}
+
+        # colors
```
Comment (grzanka) [inline pytripgui/canvas_vc/plotter/managers/voi_manager.py:22]: Use colors consistent with `pytrip` definition, see following methods of `Vdx` class:
- `define_colors`
- `get_color`
- `set_color`

Code at:
https://github.com/pytrip/pytrip/blob/e2b4d22d5ebba50e5e3545d2e140a56982234ca7/pytrip/vdx.py#L824

The reasons of keeping colors at `pytrip` level are following:
 - these colors could be used in subsequent development of `cubeslicer` code (command line tools embedded in pytrip package which generates .png files for each slice found in CT and dose cube)
 - some logic can be added to color definition, i.e. typical planning volumes (i.e. CTV, PTV) could get some typical colors, i.e. red
--

**Review by michal367 — COMMENTED** [review]
There are no contours on simulation canvas

**Review by michal367 — COMMENTED** [review]
Performance is worse:
changing item in patient tree is 2-3 times slower than before

File: pytripgui/canvas_vc/plotter/managers/voi_manager.py (hunk 2 above)
Comment (rozchlastywacz) [reply pytripgui/canvas_vc/plotter/managers/voi_manager.py]: Done for now, but needs to be changed after changes in pytrip

**Comment by grzanka** [comment]
@rozchlastywacz fix conflicts

**Comment by Yoanka** [comment]
While scrolling a plot with contours in coronal or sagittal view, the plot gets squashed into a rectangle the size of the top quarter of the view area (see image 1). Only the top image with wrong scaling is scrolling through slices, while the bigger one stays frozen. Bug occurs in both patient and dose views, as long as contours are selected and displayed (does not occur if the current slice doesn't contain a selected contour).
Rescaling the window causes a white rectangle to appear instead of the frozen second scan (see image 2).
After switching to a different patient or dose, the plot scaling goes back to normal until a contour is displayed again in sagittal or coronal view.

image
image

**Comment by rozchlastywacz** [comment]
@Yoanka 
Last commit of mine:
https://github.com/pytrip/pytripgui/pull/437/commits/8a1644d1e53ab5c62a372b7a8e8fa2cffbd12749
I`m quite aware of that fact, as commit message says:
image

**Review by grzanka — APPROVED** [review]
looks OK

**Pull request merged**